azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/secrets/KeyVaultSecretTemplate.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.keyvault.spring.secrets;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+import com.microsoft.azure.keyvault.models.SecretItem;
+import com.microsoft.azure.keyvault.spring.AbstractKeyVaultTemplate;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of {@link KeyVaultSecretOperation}
+ *
+ */
+public class KeyVaultSecretTemplate extends AbstractKeyVaultTemplate implements KeyVaultSecretOperation {
+    private final Map<String, Set<String>> secretNamesByKeyVault = new ConcurrentHashMap<>();
+    private final Map<Pair<String, String>, String> secretsByKeyVault = new ConcurrentHashMap<>();
+
+    private long lastUpdatedTime = System.currentTimeMillis();
+
+    public KeyVaultSecretTemplate(String clientId, String clientSecret) {
+        super(clientId, clientSecret);
+    }
+
+    public KeyVaultSecretTemplate(String clientId, String clientSecret, long refreshIntervalMS) {
+        super(clientId, clientSecret, refreshIntervalMS);
+    }
+
+    @Override
+    public String getSecret(String keyVaultName, String secretName) {
+        final Pair<String, String> keyVaultAndSecret = Pair.of(keyVaultName, secretName);
+        if (useCache) {
+            clearCacheAfterInterval();
```

Comment:
I suggest not to clear cache, what we really need is update new values. Just leave old values there.